From aa1f9515ec942056168241d76ad00dadafc9d797 Mon Sep 17 00:00:00 2001
From: Thomas Gerbet <thomas.gerbet@enalean.com>
Date: Thu, 26 Jan 2023 11:05:08 +0100
Subject: [PATCH] Compatibility with mPdf 8

---
 MpdfAction.php | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/extensions/Mpdf/MpdfAction.php b/extensions/Mpdf/MpdfAction.php
index bcb342a..ce16f91 100644
--- a/extensions/Mpdf/MpdfAction.php
+++ b/extensions/Mpdf/MpdfAction.php
@@ -93,7 +93,8 @@ class MpdfAction extends Action {
 					$orientation = $matches[1];
 				}
 			}
-			$mpdf = new mPDF( $mode, $format, 0, '', $marginLeft, $marginRight, $marginTop, $marginBottom, $marginHeader, $marginFooter, $orientation );
+			$orientation = in_array(strtoupper($orientation), ['P', 'L']) ? $orientation : 'P';
+			$mpdf = new \Mpdf\Mpdf( ['tempDir' => wfTempDir() . '/mdpf-extension', 'mode' => $mode, 'format' => "$format-$orientation", 'margin_left' => $marginLeft, 'margin_right' => $marginRight, 'margin_top' => $marginTop, 'margin_bottom' => $marginBottom, 'margin_header' => $marginHeader, 'margin_footer' => $marginFooter] );

 			// Suppress warning messages, because the mPDF library
 			// itself generates warnings (due to trying to add
--
2.39.1

