*** Makefile	Wed Aug 29 17:45:32 2001
--- Makefile.withgen	Wed Aug 29 17:45:14 2001
***************
*** 9,18 ****
  mandir=${prefix}/man
  
  LIBS=-lreadline -ldl  -lcrypt
! CLIENT_LIBS=@CLIENT_LIBS@
  CC=gcc
  CFLAGS=-O
! CPPFLAGS=
  LDFLAGS=
  AWK=gawk
  
--- 9,18 ----
  mandir=${prefix}/man
  
  LIBS=-lreadline -ldl  -lcrypt
! CLIENT_LIBS=
  CC=gcc
  CFLAGS=-O
! CPPFLAGS= -DDEBUG_PASSWORD=1
  LDFLAGS=
  AWK=gawk
  
***************
*** 80,86 ****
  
  SPROGS = bin/smbd bin/nmbd bin/swat
  PROGS1 = bin/smbclient bin/smbspool bin/testparm bin/testprns bin/smbstatus 
! PROGS2 = bin/rpcclient bin/smbpasswd bin/make_smbcodepage bin/make_unicodemap  
  MPROGS = 
  PROGS = $(PROGS1) $(PROGS2) $(MPROGS) bin/nmblookup bin/make_printerdef 
  
--- 80,86 ----
  
  SPROGS = bin/smbd bin/nmbd bin/swat
  PROGS1 = bin/smbclient bin/smbspool bin/testparm bin/testprns bin/smbstatus 
! PROGS2 = bin/rpcclient bin/smbpasswd bin/gensmbpasswd bin/make_smbcodepage bin/make_unicodemap  
  MPROGS = 
  PROGS = $(PROGS1) $(PROGS2) $(MPROGS) bin/nmblookup bin/make_printerdef 
  
***************
*** 208,213 ****
--- 208,216 ----
  SMBPASSWD_OBJ = utils/smbpasswd.o $(PARAM_OBJ) $(LIBSMB_OBJ) $(PASSDB_OBJ) \
                  $(UBIQX_OBJ) $(RPC_CLIENT_OBJ) $(RPC_PARSE_OBJ) $(LIB_OBJ)
  
+ GENSMBPASSWD_OBJ = utils/gensmbpasswd.o $(PARAM_OBJ) $(LIBSMB_OBJ) $(PASSDB_OBJ) \
+                 $(UBIQX_OBJ) $(RPC_CLIENT_OBJ) $(RPC_PARSE_OBJ) $(LIB_OBJ)
+ 
  RPCCLIENT_OBJ = rpcclient/rpcclient.o \
               rpcclient/display.o \
               rpcclient/cmd_lsarpc.o \
***************
*** 411,416 ****
--- 414,423 ----
  bin/smbpasswd: $(SMBPASSWD_OBJ) bin/.dummy
  	@echo Linking $@
  	@$(CC) $(FLAGS) -o $@ $(SMBPASSWD_OBJ) $(LDFLAGS) $(LIBS)
+ 
+ bin/gensmbpasswd: $(GENSMBPASSWD_OBJ) bin/.dummy
+ 	@echo Linking $@
+ 	@$(CC) $(FLAGS) -o $@ $(GENSMBPASSWD_OBJ) $(LDFLAGS) $(LIBS)
  
  bin/make_smbcodepage: $(MAKE_SMBCODEPAGE_OBJ) bin/.dummy
  	@echo Linking $@
