diff --git a/maintenance/tables.sql b/maintenance/tables.sql
index b76ad175a6..36dd573f16 100644
--- a/maintenance/tables.sql
+++ b/maintenance/tables.sql
@@ -1412,7 +1412,7 @@ CREATE TABLE /*_*/searchindex (

   -- Munged version of body text
   si_text mediumtext NOT NULL
-) ENGINE=MyISAM DEFAULT CHARSET=utf8;
+) ENGINE=InnoDB, DEFAULT CHARSET='utf8mb4';

 CREATE UNIQUE INDEX /*i*/si_page ON /*_*/searchindex (si_page);
 CREATE FULLTEXT INDEX /*i*/si_title ON /*_*/searchindex (si_title);
