--- cvsgraph.conf.orig	2005-12-18 03:07:47.000000000 +0200
+++ cvsgraph.conf	2005-12-18 22:30:41.000000000 +0200
@@ -124,8 +124,8 @@
 #	CVS/RCS repository can be found
 # cvsmodule <string>
 #
-cvsroot		= "/home/bertho/tmp/cvstest";
-cvsmodule	= "ttt";
+cvsroot		= "/var/cvs";
+cvsmodule	= "";
 
 # color_bg <color>
 #	The background color of the image
@@ -175,7 +175,7 @@
 #	Sets the error/warning message font
 msg_color	= "#800000";
 msg_font	= medium;
-msg_ttfont	= "/dos/windows/fonts/ariali.ttf";
+msg_ttfont	= "/usr/share/fonts/bitstream-vera/VeraIt.ttf";
 msg_ttsize	= 11.0;
 
 # parse_logs <boolean>
@@ -206,7 +206,7 @@
 #	tags will be shown.
 #	Note: tags matched with merge_from/merge_to will still be displayed.
 tag_font	= medium;
-tag_ttfont	= "/dos/windows/fonts/ariali.ttf";
+tag_ttfont	= "/usr/share/fonts/bitstream-vera/VeraIt.ttf";
 tag_ttsize	= 11.0;
 tag_color	= "#007000";
 #tag_ignore	= "(test|alpha)_release";
@@ -245,7 +245,7 @@
 #
 #rev_hidenumber = false;
 rev_font	= giant;
-rev_ttfont	= "/dos/windows/fonts/arial.ttf";
+rev_ttfont	= "/usr/share/fonts/bitstream-vera/Vera.ttf";
 rev_ttsize	= 12.0;
 rev_idtext	= [ "state" == "dead" "(Dead %R)" "%R" ];
 #rev_color	= "#000000";
@@ -262,7 +262,7 @@
 #rev_text	= "%d";	# or "%d\n%a, %s" for author and state too
 rev_text	= [ "state" == "dead" "Died %d\nMaybe alive in other branch" "%d" ];
 rev_text_font	= tiny;
-rev_text_ttfont	= "/dos/windows/fonts/times.ttf";
+rev_text_ttfont	= "/usr/share/fonts/bitstream-vera/VeraSe.ttf";
 rev_text_ttsize	= 9.0;
 rev_text_color	= "#500020";
 rev_maxtags	= 25;
@@ -373,11 +373,11 @@
 #	tag. This option you would normally want to set from the
 #	command line with the -O option.
 branch_font	= medium;
-branch_ttfont	= "/dos/windows/fonts/arialbd.ttf";
+branch_ttfont	= "/usr/share/fonts/bitstream-vera/VeraBd.ttf";
 branch_ttsize	= 18.0;
 branch_tag_color= "#000080";
 branch_tag_font	= medium;
-branch_tag_ttfont = "/dos/windows/fonts/arialbi.ttf";
+branch_tag_ttfont = "/usr/share/fonts/bitstream-vera/VeraBI.ttf";
 branch_tag_ttsize = 14.0;
 branch_color	= "#0000c0";
 branch_bgcolor	= "#ffffc0";
@@ -408,7 +408,7 @@
 title_x		= 10;
 title_y		= 5;
 title_font	= small;
-title_ttfont	= "/dos/windows/fonts/times.ttf";
+title_ttfont	= "/usr/share/fonts/bitstream-vera/VeraSe.ttf";
 title_ttsize	= 10.0;
 title_align	= left;
 title_color	= "#800000";
