diff -r -u mailman-2.1.5.orig/bin/Makefile.in mailman-2.1.5.build/bin/Makefile.in
--- mailman-2.1.5.orig/bin/Makefile.in	2003-12-24 12:03:15.000000000 -0500
+++ mailman-2.1.5.build/bin/Makefile.in	2004-10-15 15:36:46.000000000 -0400
@@ -28,7 +28,6 @@
 DESTDIR=
 
 CC=		@CC@
-CHMOD=  	@CHMOD@
 INSTALL=	@INSTALL@
 
 DEFS=   	@DEFS@
@@ -55,7 +54,7 @@
 # Modes for directories and executables created by the install
 # process.  Default to group-writable directories but
 # user-only-writable for executables.
-DIRMODE=	775
+DIRMODE=	2775
 EXEMODE=	755
 FILEMODE=	644
 INSTALL_PROGRAM=$(INSTALL) -m $(EXEMODE)
@@ -71,8 +70,6 @@
 	    $(INSTALL) -m $(EXEMODE) $(BUILDDIR)/$$f $(DESTDIR)$(SCRIPTSDIR); \
 	done
 
-finish:
-
 clean:
 
 distclean:
Only in mailman-2.1.5.build/bin: Makefile.in~
Only in mailman-2.1.5.build: configure
diff -r -u mailman-2.1.5.orig/cron/Makefile.in mailman-2.1.5.build/cron/Makefile.in
--- mailman-2.1.5.orig/cron/Makefile.in	2003-03-31 14:27:07.000000000 -0500
+++ mailman-2.1.5.build/cron/Makefile.in	2004-10-15 15:36:46.000000000 -0400
@@ -28,7 +28,6 @@
 DESTDIR=
 
 CC=		@CC@
-CHMOD=  	@CHMOD@
 INSTALL=	@INSTALL@
 
 DEFS=   	@DEFS@
@@ -68,8 +67,6 @@
 	    $(INSTALL) -m $(EXEMODE) $(BUILDDIR)/$$f $(DESTDIR)$(CRONDIR); \
 	done
 
-finish:
-
 clean:
 
 distclean:
Only in mailman-2.1.5.build/cron: Makefile.in~
diff -r -u mailman-2.1.5.orig/Mailman/Archiver/Makefile.in mailman-2.1.5.build/Mailman/Archiver/Makefile.in
--- mailman-2.1.5.orig/Mailman/Archiver/Makefile.in	2003-03-31 14:26:58.000000000 -0500
+++ mailman-2.1.5.build/Mailman/Archiver/Makefile.in	2004-10-15 15:36:46.000000000 -0400
@@ -28,7 +28,6 @@
 DESTDIR=
 
 CC=		@CC@
-CHMOD=  	@CHMOD@
 INSTALL=	@INSTALL@
 
 DEFS=   	@DEFS@
@@ -47,7 +46,7 @@
 # Modes for directories and executables created by the install
 # process.  Default to group-writable directories but
 # user-only-writable for executables.
-DIRMODE=	775
+DIRMODE=	2775
 EXEMODE=	755
 FILEMODE=	644
 INSTALL_PROGRAM=$(INSTALL) -m $(EXEMODE)
@@ -63,8 +62,6 @@
 	    $(INSTALL) -m $(FILEMODE) $(srcdir)/$$f $(DESTDIR)$(PACKAGEDIR); \
 	done
 
-finish:
-
 clean:
 
 distclean:
Only in mailman-2.1.5.build/Mailman/Archiver: Makefile.in~
diff -r -u mailman-2.1.5.orig/Mailman/Bouncers/Makefile.in mailman-2.1.5.build/Mailman/Bouncers/Makefile.in
--- mailman-2.1.5.orig/Mailman/Bouncers/Makefile.in	2003-03-31 14:26:58.000000000 -0500
+++ mailman-2.1.5.build/Mailman/Bouncers/Makefile.in	2004-10-15 15:36:46.000000000 -0400
@@ -28,7 +28,6 @@
 DESTDIR=
 
 CC=		@CC@
-CHMOD=  	@CHMOD@
 INSTALL=	@INSTALL@
 
 DEFS=   	@DEFS@
@@ -45,7 +44,7 @@
 # Modes for directories and executables created by the install
 # process.  Default to group-writable directories but
 # user-only-writable for executables.
-DIRMODE=	775
+DIRMODE=	2775
 EXEMODE=	755
 FILEMODE=	644
 INSTALL_PROGRAM=$(INSTALL) -m $(EXEMODE)
@@ -61,8 +60,6 @@
 	    $(INSTALL) -m $(FILEMODE) $(srcdir)/$$f $(DESTDIR)$(PACKAGEDIR); \
 	done
 
-finish:
-
 clean:
 
 distclean:
Only in mailman-2.1.5.build/Mailman/Bouncers: Makefile.in~
diff -r -u mailman-2.1.5.orig/Mailman/Cgi/Makefile.in mailman-2.1.5.build/Mailman/Cgi/Makefile.in
--- mailman-2.1.5.orig/Mailman/Cgi/Makefile.in	2003-03-31 14:26:59.000000000 -0500
+++ mailman-2.1.5.build/Mailman/Cgi/Makefile.in	2004-10-15 15:36:46.000000000 -0400
@@ -28,7 +28,6 @@
 DESTDIR=
 
 CC=		@CC@
-CHMOD=  	@CHMOD@
 INSTALL=	@INSTALL@
 
 DEFS=   	@DEFS@
@@ -47,7 +46,7 @@
 # Modes for directories and executables created by the install
 # process.  Default to group-writable directories but
 # user-only-writable for executables.
-DIRMODE=	775
+DIRMODE=	2775
 EXEMODE=	755
 FILEMODE=	644
 INSTALL_PROGRAM=$(INSTALL) -m $(EXEMODE)
@@ -63,8 +62,6 @@
 	    $(INSTALL) -m $(FILEMODE) $(srcdir)/$$f $(DESTDIR)$(CGIDIR); \
         done
 
-finish:
-
 clean:
 
 distclean:
Only in mailman-2.1.5.build/Mailman/Cgi: Makefile.in~
diff -r -u mailman-2.1.5.orig/Mailman/Commands/Makefile.in mailman-2.1.5.build/Mailman/Commands/Makefile.in
--- mailman-2.1.5.orig/Mailman/Commands/Makefile.in	2003-03-31 14:27:00.000000000 -0500
+++ mailman-2.1.5.build/Mailman/Commands/Makefile.in	2004-10-15 15:36:46.000000000 -0400
@@ -28,7 +28,6 @@
 DESTDIR=
 
 CC=		@CC@
-CHMOD=  	@CHMOD@
 INSTALL=	@INSTALL@
 
 DEFS=   	@DEFS@
@@ -45,7 +44,7 @@
 # Modes for directories and executables created by the install
 # process.  Default to group-writable directories but
 # user-only-writable for executables.
-DIRMODE=	775
+DIRMODE=	2775
 EXEMODE=	755
 FILEMODE=	644
 INSTALL_PROGRAM=$(INSTALL) -m $(EXEMODE)
@@ -61,8 +60,6 @@
 	    $(INSTALL) -m $(FILEMODE) $(srcdir)/$$f $(DESTDIR)$(PACKAGEDIR); \
 	done
 
-finish:
-
 clean:
 
 distclean:
Only in mailman-2.1.5.build/Mailman/Commands: Makefile.in~
diff -r -u mailman-2.1.5.orig/Mailman/Gui/Makefile.in mailman-2.1.5.build/Mailman/Gui/Makefile.in
--- mailman-2.1.5.orig/Mailman/Gui/Makefile.in	2003-03-31 14:27:01.000000000 -0500
+++ mailman-2.1.5.build/Mailman/Gui/Makefile.in	2004-10-15 15:36:46.000000000 -0400
@@ -28,7 +28,6 @@
 DESTDIR=
 
 CC=		@CC@
-CHMOD=  	@CHMOD@
 INSTALL=	@INSTALL@
 
 DEFS=   	@DEFS@
@@ -45,7 +44,7 @@
 # Modes for directories and executables created by the install
 # process.  Default to group-writable directories but
 # user-only-writable for executables.
-DIRMODE=	775
+DIRMODE=	2775
 EXEMODE=	755
 FILEMODE=	644
 INSTALL_PROGRAM=$(INSTALL) -m $(EXEMODE)
@@ -61,8 +60,6 @@
 	    $(INSTALL) -m $(FILEMODE) $(srcdir)/$$f $(DESTDIR)$(PACKAGEDIR); \
 	done
 
-finish:
-
 clean:
 
 distclean:
Only in mailman-2.1.5.build/Mailman/Gui: Makefile.in~
diff -r -u mailman-2.1.5.orig/Mailman/Handlers/Makefile.in mailman-2.1.5.build/Mailman/Handlers/Makefile.in
--- mailman-2.1.5.orig/Mailman/Handlers/Makefile.in	2003-03-31 14:27:01.000000000 -0500
+++ mailman-2.1.5.build/Mailman/Handlers/Makefile.in	2004-10-15 15:36:46.000000000 -0400
@@ -28,7 +28,6 @@
 DESTDIR=
 
 CC=		@CC@
-CHMOD=  	@CHMOD@
 INSTALL=	@INSTALL@
 
 DEFS=   	@DEFS@
@@ -45,7 +44,7 @@
 # Modes for directories and executables created by the install
 # process.  Default to group-writable directories but
 # user-only-writable for executables.
-DIRMODE=	775
+DIRMODE=	2775
 EXEMODE=	755
 FILEMODE=	644
 INSTALL_PROGRAM=$(INSTALL) -m $(EXEMODE)
@@ -61,8 +60,6 @@
 	    $(INSTALL) -m $(FILEMODE) $(srcdir)/$$f $(DESTDIR)$(PACKAGEDIR); \
 	done
 
-finish:
-
 clean:
 
 distclean:
Only in mailman-2.1.5.build/Mailman/Handlers: Makefile.in~
diff -r -u mailman-2.1.5.orig/Mailman/Logging/Makefile.in mailman-2.1.5.build/Mailman/Logging/Makefile.in
--- mailman-2.1.5.orig/Mailman/Logging/Makefile.in	2003-03-31 14:27:02.000000000 -0500
+++ mailman-2.1.5.build/Mailman/Logging/Makefile.in	2004-10-15 15:36:46.000000000 -0400
@@ -28,7 +28,6 @@
 DESTDIR=
 
 CC=		@CC@
-CHMOD=  	@CHMOD@
 INSTALL=	@INSTALL@
 
 DEFS=   	@DEFS@
@@ -45,7 +44,7 @@
 # Modes for directories and executables created by the install
 # process.  Default to group-writable directories but
 # user-only-writable for executables.
-DIRMODE=	775
+DIRMODE=	2775
 EXEMODE=	755
 FILEMODE=	644
 INSTALL_PROGRAM=$(INSTALL) -m $(EXEMODE)
@@ -61,8 +60,6 @@
 	    $(INSTALL) -m $(FILEMODE) $(srcdir)/$$f $(DESTDIR)$(PACKAGEDIR); \
 	done
 
-finish:
-
 clean:
 
 distclean:
Only in mailman-2.1.5.build/Mailman/Logging: Makefile.in~
diff -r -u mailman-2.1.5.orig/Mailman/Makefile.in mailman-2.1.5.build/Mailman/Makefile.in
--- mailman-2.1.5.orig/Mailman/Makefile.in	2003-03-31 14:26:58.000000000 -0500
+++ mailman-2.1.5.build/Mailman/Makefile.in	2004-10-15 15:36:46.000000000 -0400
@@ -20,8 +20,6 @@
 
 # Variables set by configure
 
-VERSION=	@VERSION@
-
 VPATH=		@srcdir@
 srcdir= 	@srcdir@
 bindir= 	@bindir@
@@ -30,7 +28,6 @@
 DESTDIR=
 
 CC=		@CC@
-CHMOD=  	@CHMOD@
 INSTALL=	@INSTALL@
 
 DEFS=   	@DEFS@
@@ -48,7 +45,7 @@
 # Modes for directories and executables created by the install
 # process.  Default to group-writable directories but
 # user-only-writable for executables.
-DIRMODE=	775
+DIRMODE=	2775
 EXEMODE=	755
 FILEMODE=	644
 INSTALL_PROGRAM=$(INSTALL) -m $(EXEMODE)
@@ -79,12 +76,6 @@
 	    (cd $$d; $(MAKE) DESTDIR=$(DESTDIR) install); \
 	done
 
-finish:
-	@for d in $(SUBDIRS); \
-	do \
-	    (cd $$d; $(MAKE) DESTDIR=$(DESTDIR) finish); \
-	done
-
 clean:
 	for d in $(SUBDIRS); \
 	do \
Only in mailman-2.1.5.build/Mailman: Makefile.in~
diff -r -u mailman-2.1.5.orig/Mailman/MTA/Makefile.in mailman-2.1.5.build/Mailman/MTA/Makefile.in
--- mailman-2.1.5.orig/Mailman/MTA/Makefile.in	2003-03-31 14:27:02.000000000 -0500
+++ mailman-2.1.5.build/Mailman/MTA/Makefile.in	2004-10-15 15:36:46.000000000 -0400
@@ -28,7 +28,6 @@
 DESTDIR=
 
 CC=		@CC@
-CHMOD=  	@CHMOD@
 INSTALL=	@INSTALL@
 
 DEFS=   	@DEFS@
@@ -45,7 +44,7 @@
 # Modes for directories and executables created by the install
 # process.  Default to group-writable directories but
 # user-only-writable for executables.
-DIRMODE=	775
+DIRMODE=	2775
 EXEMODE=	755
 FILEMODE=	644
 INSTALL_PROGRAM=$(INSTALL) -m $(EXEMODE)
@@ -61,8 +60,6 @@
 	    $(INSTALL) -m $(FILEMODE) $(srcdir)/$$f $(DESTDIR)$(PACKAGEDIR); \
 	done
 
-finish:
-
 clean:
 
 distclean:
Only in mailman-2.1.5.build/Mailman/MTA: Makefile.in~
diff -r -u mailman-2.1.5.orig/Mailman/Queue/Makefile.in mailman-2.1.5.build/Mailman/Queue/Makefile.in
--- mailman-2.1.5.orig/Mailman/Queue/Makefile.in	2003-03-31 14:27:03.000000000 -0500
+++ mailman-2.1.5.build/Mailman/Queue/Makefile.in	2004-10-15 15:36:46.000000000 -0400
@@ -28,7 +28,6 @@
 DESTDIR=
 
 CC=		@CC@
-CHMOD=  	@CHMOD@
 INSTALL=	@INSTALL@
 
 DEFS=   	@DEFS@
@@ -45,7 +44,7 @@
 # Modes for directories and executables created by the install
 # process.  Default to group-writable directories but
 # user-only-writable for executables.
-DIRMODE=	775
+DIRMODE=	2775
 EXEMODE=	755
 FILEMODE=	644
 INSTALL_PROGRAM=$(INSTALL) -m $(EXEMODE)
@@ -61,8 +60,6 @@
 	    $(INSTALL) -m $(FILEMODE) $(srcdir)/$$f $(DESTDIR)$(PACKAGEDIR); \
 	done
 
-finish:
-
 clean:
 
 distclean:
Only in mailman-2.1.5.build/Mailman/Queue: Makefile.in~
Only in mailman-2.1.5.orig: mailman-FHS.patch
diff -r -u mailman-2.1.5.orig/Makefile.in mailman-2.1.5.build/Makefile.in
--- mailman-2.1.5.orig/Makefile.in	2003-03-31 14:26:57.000000000 -0500
+++ mailman-2.1.5.build/Makefile.in	2004-10-15 16:26:03.000000000 -0400
@@ -61,11 +61,10 @@
 # Modes for directories and executables created by the install
 # process.  Default to group-writable directories but
 # user-only-writable for executables.
-DIRMODE=	775
+DIRMODE=	2775
 EXEMODE=	755
 FILEMODE=	644
 INSTALL_PROGRAM=$(INSTALL) -m $(EXEMODE)
-DIRSETGID=	chmod g+s
 
 DATE = $(shell python -c 'import time; print time.strftime("%d-%b-%Y"),')
 LANGPACK = README-I18N.en templates messages
@@ -85,14 +84,24 @@
 
 doinstall: $(SUBDIRS)
 	@echo "Creating architecture independent directories..."
+	dir=$(DESTDIR)$(prefix); \
+	if test ! -d $$dir; then \
+		echo "Creating directory hierarchy $$dir"; \
+		$(INSTALL) -d -m $(DIRMODE) $$dir; \
+	else true; \
+	fi;
+	dir=$(DESTDIR)$(var_prefix); \
+	if test ! -d $$dir; then \
+		echo "Creating directory hierarchy $$dir"; \
+		$(INSTALL) -d -m $(DIRMODE) $$dir; \
+	else true; \
+	fi;
 	@for d in $(VAR_DIRS); \
 	do \
 	    dir=$(DESTDIR)$(var_prefix)/$$d; \
 	    if test ! -d $$dir; then \
 		echo "Creating directory hierarchy $$dir"; \
-		$(srcdir)/mkinstalldirs $$dir; \
-		chmod $(DIRMODE) $$dir; \
-		$(DIRSETGID) $$dir; \
+		$(INSTALL) -d -m $(DIRMODE) $$dir; \
 	    else true; \
 	    fi; \
 	done
@@ -102,9 +111,7 @@
 	    dir=$(DESTDIR)$(prefix)/$$d; \
 	    if test ! -d $$dir; then \
 		echo "Creating directory hierarchy $$dir"; \
-		$(srcdir)/mkinstalldirs $$dir; \
-		chmod $(DIRMODE) $$dir; \
-		$(DIRSETGID) $$dir; \
+		$(INSTALL) -d -m $(DIRMODE) $$dir; \
 	    else true; \
 	    fi; \
 	done
@@ -114,9 +121,7 @@
 	    dir=$(DESTDIR)$(exec_prefix)/$$d; \
 	    if test ! -d $$dir; then \
 		echo "Creating directory hierarchy $$dir"; \
-		$(srcdir)/mkinstalldirs $$dir; \
-		chmod $(DIRMODE) $$dir; \
-		$(DIRSETGID) $$dir; \
+		$(INSTALL) -d -m $(DIRMODE) $$dir; \
 	    else true; \
 	    fi; \
 	done
Only in mailman-2.1.5.build: Makefile.in~
diff -r -u mailman-2.1.5.orig/messages/Makefile.in mailman-2.1.5.build/messages/Makefile.in
--- mailman-2.1.5.orig/messages/Makefile.in	2004-04-24 22:30:04.000000000 -0400
+++ mailman-2.1.5.build/messages/Makefile.in	2004-10-15 15:36:46.000000000 -0400
@@ -32,7 +32,6 @@
 DESTDIR=
 
 CC=		@CC@
-CHMOD=  	@CHMOD@
 INSTALL=	@INSTALL@
 
 DEFS=   	@DEFS@
@@ -43,7 +42,6 @@
 CFLAGS=		$(OPT) $(DEFS)
 PACKAGEDIR= 	$(prefix)/messages
 SHELL=		/bin/sh
-DIRSETGID=	chmod g+s
 MSGFMT= 	@PYTHON@ ../build/bin/msgfmt.py
 MSGMERGE=	msgmerge
 
@@ -60,7 +58,7 @@
 # Modes for directories and executables created by the install
 # process.  Default to group-writable directories but
 # user-only-writable for executables.
-DIRMODE=	775
+DIRMODE=	2775
 EXEMODE=	755
 FILEMODE=	644
 INSTALL_PROGRAM=$(INSTALL) -m $(EXEMODE)
@@ -90,9 +88,8 @@
 	do \
 	    dir=$(DESTDIR)$(prefix)/$$d; \
 	    echo "Creating language directory $$dir"; \
-	    $(srcdir)/../mkinstalldirs $$dir; \
-	    chmod $(DIRMODE) $$dir; \
-	    $(DIRSETGID) $$dir; \
+	    $(INSTALL) -d -m $(DIRMODE) `dirname $$dir`; \
+	    $(INSTALL) -d -m $(DIRMODE) $$dir; \
 	done
 	@for d in $(LANGUAGES); \
 	do \
@@ -112,8 +109,6 @@
 
 mofiles: $(MOFILES)
 
-finish:
-
 clean:
 	-rm -f */LC_MESSAGES/mailman.mo
 
Only in mailman-2.1.5.build/messages: Makefile.in~
diff -r -u mailman-2.1.5.orig/misc/Makefile.in mailman-2.1.5.build/misc/Makefile.in
--- mailman-2.1.5.orig/misc/Makefile.in	2004-05-13 23:34:34.000000000 -0400
+++ mailman-2.1.5.build/misc/Makefile.in	2004-10-15 15:36:46.000000000 -0400
@@ -29,7 +29,6 @@
 DESTDIR=
 
 CC=		@CC@
-CHMOD=  	@CHMOD@
 INSTALL=	@INSTALL@
 PYTHON=		@PYTHON@
 
@@ -61,7 +60,7 @@
 # Modes for directories and executables created by the install
 # process.  Default to group-writable directories but
 # user-only-writable for executables.
-DIRMODE=	775
+DIRMODE=	2775
 EXEMODE=	755
 FILEMODE=	644
 DATAMODE=	664
@@ -93,8 +92,6 @@
 	    (cd $(PKGDIR)/$$p ; umask 02 ; PYTHONPATH=$(PYTHONLIBDIR) $(PYTHON) $(SETUPCMD)); \
 	done
 
-finish:
-
 clean:
 
 distclean:
Only in mailman-2.1.5.build/misc: Makefile.in~
diff -r -u mailman-2.1.5.orig/scripts/Makefile.in mailman-2.1.5.build/scripts/Makefile.in
--- mailman-2.1.5.orig/scripts/Makefile.in	2003-03-31 14:27:12.000000000 -0500
+++ mailman-2.1.5.build/scripts/Makefile.in	2004-10-15 15:36:46.000000000 -0400
@@ -28,7 +28,6 @@
 DESTDIR=
 
 CC=		@CC@
-CHMOD=  	@CHMOD@
 INSTALL=	@INSTALL@
 
 DEFS=   	@DEFS@
@@ -49,7 +48,7 @@
 # Modes for directories and executables created by the install
 # process.  Default to group-writable directories but
 # user-only-writable for executables.
-DIRMODE=	775
+DIRMODE=	2775
 EXEMODE=	755
 FILEMODE=	644
 INSTALL_PROGRAM=$(INSTALL) -m $(EXEMODE)
@@ -68,8 +67,6 @@
 	$(INSTALL) -m $(FILEMODE) $(srcdir)/join    $(DESTDIR)$(SCRIPTSDIR)/subscribe
 	$(INSTALL) -m $(FILEMODE) $(srcdir)/leave   $(DESTDIR)$(SCRIPTSDIR)/unsubscribe
 
-finish:
-
 clean:
 
 distclean:
Only in mailman-2.1.5.build/scripts: Makefile.in~
diff -r -u mailman-2.1.5.orig/src/Makefile.in mailman-2.1.5.build/src/Makefile.in
--- mailman-2.1.5.orig/src/Makefile.in	2003-03-31 14:27:14.000000000 -0500
+++ mailman-2.1.5.build/src/Makefile.in	2004-10-15 15:36:46.000000000 -0400
@@ -28,7 +28,6 @@
 DESTDIR=
 
 CC=		@CC@
-CHMOD=  	@CHMOD@
 INSTALL=	@INSTALL@
 PYTHON=  	@PYTHON@
 
@@ -63,10 +62,9 @@
 # Modes for directories and executables created by the install
 # process.  Default to group-writable directories but
 # user-only-writable for executables.
-DIRMODE=	775
-EXEMODE=	755
+DIRMODE=	2775
+EXEMODE=	2755
 INSTALL_PROGRAM=$(INSTALL) -m $(EXEMODE)
-DIRSETGID=	chmod g+s
 
 # Fixed definitions
 
@@ -110,20 +108,10 @@
 	do \
 	    exe=$(DESTDIR)$(CGIDIR)/$$f$(CGIEXT); \
 	    $(INSTALL_PROGRAM) $$f $$exe; \
-	    $(DIRSETGID) $$exe; \
 	done
 	for f in $(MAIL_PROGS); \
 	do \
 	    $(INSTALL_PROGRAM) $$f $(DESTDIR)$(MAILDIR); \
-	    $(DIRSETGID) $(DESTDIR)$(MAILDIR)/$$f; \
-	done
-
-finish:
-	-for f in $(SUID_CGI_PROGS); \
-	do \
-	    exe=$(DESTDIR)$(CGIDIR)/$$f$(CGIEXT); \
-	    chown $(MAILMAN_USER) $$exe; \
-	    chmod u+s $$exe; \
 	done
 
 clean:
Only in mailman-2.1.5.build/src: Makefile.in~
diff -r -u mailman-2.1.5.orig/templates/Makefile.in mailman-2.1.5.build/templates/Makefile.in
--- mailman-2.1.5.orig/templates/Makefile.in	2004-04-24 22:30:04.000000000 -0400
+++ mailman-2.1.5.build/templates/Makefile.in	2004-10-18 12:49:01.000000000 -0400
@@ -28,7 +28,6 @@
 DESTDIR=
 
 CC=		@CC@
-CHMOD=  	@CHMOD@
 INSTALL=	@INSTALL@
 TRUE=		@TRUE@
 
@@ -48,7 +47,7 @@
 # Modes for directories and executables created by the install
 # process.  Default to group-writable directories but
 # user-only-writable for executables.
-DIRMODE=	775
+DIRMODE=	2775
 EXEMODE=	755
 FILEMODE=	644
 INSTALL_PROGRAM=$(INSTALL) -m $(EXEMODE)
@@ -61,15 +60,13 @@
 install:
 	for d in $(LANGUAGES); \
 	do \
-	    $(srcdir)/../mkinstalldirs $(DESTDIR)$(TEMPLATEDIR)/$$d; \
+	    $(INSTALL) -d -m $(DIRMODE) $(DESTDIR)$(TEMPLATEDIR)/$$d; \
 	    for f in $(srcdir)/$$d/*.html $(srcdir)/$$d/*.txt; \
 	    do \
 	    	$(INSTALL) -m $(FILEMODE) $$f $(DESTDIR)$(TEMPLATEDIR)/$$d; \
 	    done; \
 	done
 
-finish:
-
 clean:
 
 distclean:
Only in mailman-2.1.5.build/templates: Makefile.in~
diff -r -u mailman-2.1.5.orig/tests/bounces/Makefile.in mailman-2.1.5.build/tests/bounces/Makefile.in
--- mailman-2.1.5.orig/tests/bounces/Makefile.in	2003-03-31 14:27:16.000000000 -0500
+++ mailman-2.1.5.build/tests/bounces/Makefile.in	2004-10-15 15:36:46.000000000 -0400
@@ -28,7 +28,6 @@
 DESTDIR=
 
 CC=		@CC@
-CHMOD=  	@CHMOD@
 INSTALL=	@INSTALL@
 
 DEFS=   	@DEFS@
@@ -46,7 +45,7 @@
 # Modes for directories and executables created by the install
 # process.  Default to group-writable directories but
 # user-only-writable for executables.
-DIRMODE=	775
+DIRMODE=	2775
 EXEMODE=	755
 FILEMODE=	644
 INSTALL_PROGRAM=$(INSTALL) -m $(EXEMODE)
@@ -60,8 +59,6 @@
 	    $(INSTALL) -m $(FILEMODE) $$f $(DESTDIR)$(BOUNCEDIR); \
 	done
 
-finish:
-
 clean:
 
 distclean:
Only in mailman-2.1.5.build/tests/bounces: Makefile.in~
diff -r -u mailman-2.1.5.orig/tests/Makefile.in mailman-2.1.5.build/tests/Makefile.in
--- mailman-2.1.5.orig/tests/Makefile.in	2003-03-31 14:27:15.000000000 -0500
+++ mailman-2.1.5.build/tests/Makefile.in	2004-10-15 15:36:46.000000000 -0400
@@ -28,7 +28,6 @@
 DESTDIR=
 
 CC=		@CC@
-CHMOD=  	@CHMOD@
 INSTALL=	@INSTALL@
 
 DEFS=   	@DEFS@
@@ -46,7 +45,7 @@
 # Modes for directories and executables created by the install
 # process.  Default to group-writable directories but
 # user-only-writable for executables.
-DIRMODE=	775
+DIRMODE=	2775
 EXEMODE=	755
 FILEMODE=	644
 INSTALL_PROGRAM=$(INSTALL) -m $(EXEMODE)
@@ -71,8 +70,6 @@
 	    (cd $$d; $(MAKE) DESTDIR=$(DESTDIR) install); \
 	done
 
-finish:
-
 clean:
 
 distclean:
Only in mailman-2.1.5.build/tests: Makefile.in~
diff -r -u mailman-2.1.5.orig/tests/msgs/Makefile.in mailman-2.1.5.build/tests/msgs/Makefile.in
--- mailman-2.1.5.orig/tests/msgs/Makefile.in	2003-03-31 14:27:17.000000000 -0500
+++ mailman-2.1.5.build/tests/msgs/Makefile.in	2004-10-15 15:36:46.000000000 -0400
@@ -28,7 +28,6 @@
 DESTDIR=
 
 CC=		@CC@
-CHMOD=  	@CHMOD@
 INSTALL=	@INSTALL@
 
 DEFS=   	@DEFS@
@@ -46,7 +45,7 @@
 # Modes for directories and executables created by the install
 # process.  Default to group-writable directories but
 # user-only-writable for executables.
-DIRMODE=	775
+DIRMODE=	2775
 EXEMODE=	755
 FILEMODE=	644
 INSTALL_PROGRAM=$(INSTALL) -m $(EXEMODE)
@@ -60,8 +59,6 @@
 	    $(INSTALL) -m $(FILEMODE) $$f $(DESTDIR)$(MSGSDIR); \
 	done
 
-finish:
-
 clean:
 
 distclean:
Only in mailman-2.1.5.build/tests/msgs: Makefile.in~
