diff -up mailman-2.1.9/misc/mailman.in.init-directrun mailman-2.1.9/misc/mailman.in
--- mailman-2.1.9/misc/mailman.in.init-directrun	2007-12-11 14:27:08.000000000 +0100
+++ mailman-2.1.9/misc/mailman.in	2007-12-11 14:27:54.000000000 +0100
@@ -32,7 +32,6 @@
 # config: @prefix@/Mailman/mm_cfg.py
 # pidfile: @PID_DIR@/master-qrunner.pid
 
-PYTHON=@PYTHON@
 MAILMANHOME=@prefix@
 MAILMANCTL=$MAILMANHOME/bin/mailmanctl
 
@@ -75,7 +74,7 @@ prog="mailman"
 function start()
 {
     echo -n $"Starting $prog: "
-    daemon $PYTHON $MAILMANCTL -s -q start
+    daemon $MAILMANCTL -s -q start
     RETVAL=$?
     if [ $RETVAL -eq 0 ]
     then
@@ -89,7 +88,7 @@ function start()
 function stop()
 {
     echo -n $"Shutting down $prog: "
-    daemon $PYTHON $MAILMANCTL -q stop
+    daemon $MAILMANCTL -q stop
     RETVAL=$?
     if [ $RETVAL -eq 0 ]
     then
@@ -125,7 +124,7 @@ case "$1" in
     ;;
 
 'condrestart')
-    $PYTHON $MAILMANCTL -q -u status
+    $MAILMANCTL -q -u status
     retval=$?
     if [ $retval -eq 0 ]
     then
@@ -135,7 +134,7 @@ case "$1" in
     ;;
 
 'status')
-    $PYTHON $MAILMANCTL -u status
+    $MAILMANCTL -u status
     RETVAL=$?
     ;;
 
