Only in mailman-2.1.5.FHS: autom4te.cache
diff -r -u mailman-2.1.5.orig/bin/check_perms mailman-2.1.5.FHS/bin/check_perms
--- mailman-2.1.5.orig/bin/check_perms	2003-03-31 15:07:55.000000000 -0500
+++ mailman-2.1.5.FHS/bin/check_perms	2004-10-08 16:05:09.000000000 -0400
@@ -164,7 +164,8 @@
         print _('checking mode for %(prefix)s')
     dirs = {}
     for d in (mm_cfg.PREFIX, mm_cfg.EXEC_PREFIX, mm_cfg.VAR_PREFIX,
-              mm_cfg.LOG_DIR):
+              mm_cfg.CONFIG_DIR, mm_cfg.DATA_DIR, mm_cfg.LOCK_DIR,
+              mm_cfg.LOG_DIR, mm_cfg.QUEUE_DIR, mm_cfg.PID_DIR):
         dirs[d] = True
     for d in dirs.keys():
         try:
Only in mailman-2.1.5.FHS/bin: check_perms~
Only in mailman-2.1.5.FHS: config.log
Only in mailman-2.1.5.FHS: configure
diff -r -u mailman-2.1.5.orig/configure.in mailman-2.1.5.FHS/configure.in
--- mailman-2.1.5.orig/configure.in	2003-12-24 12:11:48.000000000 -0500
+++ mailman-2.1.5.FHS/configure.in	2004-09-30 16:13:56.000000000 -0400
@@ -180,7 +180,7 @@
 AC_SUBST(VAR_PREFIX)
 AC_MSG_CHECKING(for --with-var-prefix)
 AC_ARG_WITH(var-prefix, dnl
-[  --with-var-prefix     directory for mutable data [/var/mailman]])
+[  --with-var-prefix    directory for mutable data [/var/mailman]])
 case "$with_var_prefix" in
         yes)    VAR_PREFIX="$default_var_prefix"; ans=$VAR_PREFIX;;
         ""|no)  VAR_PREFIX="$prefix"; ans="no";;
@@ -207,6 +207,61 @@
     prefixcheck=$VAR_PREFIX
 fi
 
+# Get the configuration file directory
+AC_SUBST(CONFIG_DIR)
+AC_MSG_CHECKING(for --with-config-dir)
+AC_ARG_WITH(config-dir, dnl
+[  --with-config-dir    specify directory for configuration data other than [VAR_]PREFIX/data])
+case "$with_config_dir" in
+    yes|no|"") CONFIG_DIR="$VAR_PREFIX/data";;
+    *)         CONFIG_DIR=$with_config_dir;;
+esac
+AC_MSG_RESULT($CONFIG_DIR)
+
+# Get the lock directory
+AC_SUBST(LOCK_DIR)
+AC_MSG_CHECKING(for --with-lock-dir)
+AC_ARG_WITH(lock-dir, dnl
+[  --with-lock-dir    specify directory for lock files other than [VAR_]PREFIX/locks])
+case "$with_lock_dir" in
+    yes|no|"") LOCK_DIR="$VAR_PREFIX/locks";;
+    *)         LOCK_DIR=$with_lock_dir;;
+esac
+AC_MSG_RESULT($LOCK_DIR)
+
+# Get the log directory
+AC_SUBST(LOG_DIR)
+AC_MSG_CHECKING(for --with-log-dir)
+AC_ARG_WITH(log-dir, dnl
+[  --with-log-dir    specify directory for log files other than [VAR_]PREFIX/logs])
+case "$with_log_dir" in
+    yes|no|"") LOG_DIR="$VAR_PREFIX/logs";;
+    *)         LOG_DIR=$with_log_dir;;
+esac
+AC_MSG_RESULT($LOG_DIR)
+
+# Get the pid directory
+AC_SUBST(PID_DIR)
+AC_MSG_CHECKING(for --with-pid-dir)
+AC_ARG_WITH(pid-dir, dnl
+[  --with-pid-dir    specify directory for the pid file other than [VAR_]PREFIX/data])
+case "$with_pid_dir" in
+    yes|no|"") PID_DIR="$VAR_PREFIX/data";;
+    *)         PID_DIR=$with_pid_dir;;
+esac
+AC_MSG_RESULT($PID_DIR)
+
+# Get the queue directory
+AC_SUBST(QUEUE_DIR)
+AC_MSG_CHECKING(for --with-queue-dir)
+AC_ARG_WITH(queue-dir, dnl
+[  --with-queue-dir    specify directory for queue files other than [VAR_]PREFIX/qfiles])
+case "$with_queue_dir" in
+    yes|no|"") QUEUE_DIR="$VAR_PREFIX/qfiles";;
+    *)         QUEUE_DIR=$with_queue_dir;;
+esac
+AC_MSG_RESULT($QUEUE_DIR)
+
 # new macro for finding group names
 AC_DEFUN(MM_FIND_GROUP_NAME, [
 # $1 == variable name
@@ -619,7 +674,7 @@
            templates/Makefile cron/Makefile scripts/Makefile messages/Makefile
            cron/crontab.in misc/mailman Makefile
            tests/Makefile tests/bounces/Makefile tests/msgs/Makefile
-           $SCRIPTS],
+           $SCRIPTS ],
         echo "configuration completed at" `date`)
 
 # Make sure all the build scripts are executable.
Only in mailman-2.1.5.FHS: configure.in~
diff -r -u mailman-2.1.5.orig/Mailman/Defaults.py.in mailman-2.1.5.FHS/Mailman/Defaults.py.in
--- mailman-2.1.5.orig/Mailman/Defaults.py.in	2004-04-24 22:30:03.000000000 -0400
+++ mailman-2.1.5.FHS/Mailman/Defaults.py.in	2004-10-08 14:38:57.000000000 -0400
@@ -1198,9 +1198,11 @@
 
 # Useful directories
 LIST_DATA_DIR   = os.path.join(VAR_PREFIX, 'lists')
-LOG_DIR         = os.path.join(VAR_PREFIX, 'logs')
-LOCK_DIR        = os.path.join(VAR_PREFIX, 'locks')
+LOG_DIR         = '@LOG_DIR@'
+LOCK_DIR        = '@LOCK_DIR@'
+CONFIG_DIR      = '@CONFIG_DIR@'
 DATA_DIR        = os.path.join(VAR_PREFIX, 'data')
+PID_DIR         = '@PID_DIR@'
 SPAM_DIR        = os.path.join(VAR_PREFIX, 'spam')
 WRAPPER_DIR     = os.path.join(EXEC_PREFIX, 'mail')
 BIN_DIR         = os.path.join(PREFIX, 'bin')
@@ -1211,7 +1213,7 @@
 PRIVATE_ARCHIVE_FILE_DIR = os.path.join(VAR_PREFIX, 'archives', 'private')
 
 # Directories used by the qrunner subsystem
-QUEUE_DIR       = os.path.join(VAR_PREFIX, 'qfiles')
+QUEUE_DIR       = '@QUEUE_DIR@'
 INQUEUE_DIR     = os.path.join(QUEUE_DIR, 'in')
 OUTQUEUE_DIR    = os.path.join(QUEUE_DIR, 'out')
 CMDQUEUE_DIR    = os.path.join(QUEUE_DIR, 'commands')
@@ -1225,9 +1227,9 @@
 MAILDIR_DIR     = os.path.join(QUEUE_DIR, 'maildir')
 
 # Other useful files
-PIDFILE = os.path.join(DATA_DIR, 'master-qrunner.pid')
-SITE_PW_FILE = os.path.join(DATA_DIR, 'adm.pw')
-LISTCREATOR_PW_FILE = os.path.join(DATA_DIR, 'creator.pw')
+PIDFILE = os.path.join(PID_DIR, 'master-qrunner.pid')
+SITE_PW_FILE = os.path.join(CONFIG_DIR, 'adm.pw')
+LISTCREATOR_PW_FILE = os.path.join(CONFIG_DIR, 'creator.pw')
 
 # Import a bunch of version numbers
 from Version import *
Only in mailman-2.1.5.FHS/Mailman: Defaults.py.in~
diff -r -u mailman-2.1.5.orig/Mailman/MTA/Postfix.py mailman-2.1.5.FHS/Mailman/MTA/Postfix.py
--- mailman-2.1.5.orig/Mailman/MTA/Postfix.py	2003-03-31 16:49:43.000000000 -0500
+++ mailman-2.1.5.FHS/Mailman/MTA/Postfix.py	2004-10-08 16:02:20.000000000 -0400
@@ -32,8 +32,8 @@
 from Mailman.Logging.Syslog import syslog
 
 LOCKFILE = os.path.join(mm_cfg.LOCK_DIR, 'creator')
-ALIASFILE = os.path.join(mm_cfg.DATA_DIR, 'aliases')
-VIRTFILE = os.path.join(mm_cfg.DATA_DIR, 'virtual-mailman')
+ALIASFILE = os.path.join(mm_cfg.CONFIG_DIR, 'aliases')
+VIRTFILE = os.path.join(mm_cfg.CONFIG_DIR, 'virtual-mailman')
 
 try:
     True, False
Only in mailman-2.1.5.FHS/Mailman/MTA: Postfix.py~
Only in mailman-2.1.5.orig: mailman-FHS.patch
diff -r -u mailman-2.1.5.orig/Makefile.in mailman-2.1.5.FHS/Makefile.in
--- mailman-2.1.5.orig/Makefile.in	2003-03-31 14:26:57.000000000 -0500
+++ mailman-2.1.5.FHS/Makefile.in	2004-10-15 16:48:17.000000000 -0400
@@ -28,6 +28,11 @@
 prefix=		@prefix@
 exec_prefix=	@exec_prefix@
 var_prefix=	@VAR_PREFIX@
+configdir=	@CONFIG_DIR@
+lockdir=	@LOCK_DIR@
+logdir=		@LOG_DIR@
+piddir=		@PID_DIR@
+queuedir=	@QUEUE_DIR@
 DESTDIR=
 
 CC=		@CC@
@@ -41,8 +46,12 @@
 OPT=		@OPT@
 CFLAGS=		@CFLAGS@ $(OPT) $(DEFS)
 
+FHS_DIRS= \
+	${configdir} ${lockdir} ${logdir} ${piddir} ${queuedir}
+
+
 VAR_DIRS= \
-	logs archives lists locks data spam qfiles \
+	archives lists data spam \
 	archives/private archives/public
 
 ARCH_INDEP_DIRS= \
@@ -96,6 +105,15 @@
 	    else true; \
 	    fi; \
 	done
+	@for d in $(FHS_DIRS); \
+	do \
+	    dir=$(DESTDIR)/$$d; \
+	    if test ! -d $$dir; then \
+		echo "Creating directory $$dir"; \
+		$(INSTALL) -d -m $(DIRMODE) $$dir; \
+	    else true; \
+	    fi; \
+	done
 	chmod o-r $(DESTDIR)$(var_prefix)/archives/private
 	@for d in $(ARCH_INDEP_DIRS); \
 	do \
Only in mailman-2.1.5.FHS: Makefile.in~
Only in mailman-2.1.5.FHS/messages: Makefile.in~
diff -r -u mailman-2.1.5.orig/misc/mailman.in mailman-2.1.5.FHS/misc/mailman.in
--- mailman-2.1.5.orig/misc/mailman.in	2003-09-25 18:13:26.000000000 -0400
+++ mailman-2.1.5.FHS/misc/mailman.in	2004-10-06 16:15:28.000000000 -0400
@@ -24,13 +24,13 @@
 # On Debian, type "update-rc.d mailman defaults"
 # On RedHat, and derivatives, install with "chkconfig --add mailman"
 #
-# chkconfig: 2345 98 12
+# chkconfig: - 98 12
 # description: Mailman is the GNU Mailing List Manager, a program that \
 #              manages electronic mail discussion groups.  For more \
 #              on GNU Mailman see http://www.list.org
 # processname: mailmanctl
 # config: @prefix@/Mailman/mm_cfg.py
-# pidfile: @prefix@/data/master-qrunner.pid
+# pidfile: @PID_DIR@/master-qrunner.pid
 
 PYTHON=@PYTHON@
 MAILMANHOME=@prefix@
Only in mailman-2.1.5.FHS/misc: mailman.in~
diff -r -u mailman-2.1.5.orig/misc/Makefile.in mailman-2.1.5.FHS/misc/Makefile.in
--- mailman-2.1.5.orig/misc/Makefile.in	2004-05-13 23:34:34.000000000 -0400
+++ mailman-2.1.5.FHS/misc/Makefile.in	2004-10-13 14:00:19.000000000 -0400
@@ -26,6 +26,12 @@
 prefix=   	@prefix@
 exec_prefix=	@exec_prefix@
 var_prefix=	@VAR_PREFIX@
+configdir=	@CONFIG_DIR@
+lockdir=	@LOCK_DIR@
+logdir=		@LOG_DIR@
+piddir=		@PID_DIR@
+queuedir=	@QUEUE_DIR@
+MAILMAN_GROUP=  @MAILMAN_GROUP@
 DESTDIR=
 
 CC=		@CC@
@@ -84,7 +90,7 @@
 	    $(INSTALL) -m $(FILEMODE) paths.py $$dir; \
 	done
 	$(INSTALL) -m $(EXEMODE) mailman $(DESTDIR)$(SCRIPTSDIR)
-	$(INSTALL) -m $(FILEMODE) sitelist.cfg $(DESTDIR)$(DATADIR)
+	$(INSTALL) -m $(FILEMODE) sitelist.cfg $(DESTDIR)$(configdir)
 
 install-packages:
 	for p in $(PACKAGES); \
Only in mailman-2.1.5.FHS/misc: Makefile.in~
Only in mailman-2.1.5.FHS/templates: Makefile.in~
