*** contrib/client-side/svn_load_dirs/svn_load_dirs.pl.in	2010-01-12 13:34:27.000000000 +0100
--- contrib/client-side/svn_load_dirs/svn_load_dirs.pl.in.new	2010-01-12 13:34:49.000000000 +0100
***************
*** 19,25 ****
  use Digest::MD5  2.20;
  use File::Copy   2.03;
  use File::Find;
! use File::Path   1.0404;
  use File::Temp   0.12   qw(tempdir tempfile);
  use Getopt::Long 2.25;
  use Text::Wrap;
--- 19,25 ----
  use Digest::MD5  2.20;
  use File::Copy   2.03;
  use File::Find;
! use File::Path;
  use File::Temp   0.12   qw(tempdir tempfile);
  use Getopt::Long 2.25;
  use Text::Wrap;
