--- FusionForge.php.renamed	2012-08-06 11:35:42.000000000 +0200
+++ FusionForge.php	2012-08-07 10:52:00.000000000 +0200
@@ -23,11 +23,31 @@
 	var $skinname = 'fusionforge', $stylename = 'fusionforge',
 		$template = 'FusionForgeTemplate', $useHeadElement = true;
 
+	function setupTemplate( $classname, $repository = false, $cache_dir = false ) {
+		$tc = new $classname();
+
+		$tc->params = array();
+		if (($tc->project = $project =
+		    group_get_object_by_name($GLOBALS['fusionforgeproject']))) {
+			$tc->params['group'] = $GLOBALS['group_id'] =
+			    $project->getID();
+			$tc->params['toptab'] = 'mediawiki';
+		}
+
+		return $tc;
+	}
+
 	/**
 	 * @param $out OutputPage
 	 */
 	function setupSkinUserCss( OutputPage $out ) {
 		global $wgHandheldStyle;
+
+		/* add FusionForge styles */
+		foreach ($GLOBALS['HTML']->stylesheets as $sheet) {
+			$out->addStyle($sheet['css'], $sheet['media']);
+		}
+
 		parent::setupSkinUserCss( $out );
 
 		$out->addModuleStyles( 'skins.monobook' );
@@ -51,6 +71,8 @@
  */
 class FusionForgeTemplate extends BaseTemplate {
 
+	var $project = false;
+
 	/**
 	 * Template filter callback for FusionForge skin.
 	 * Takes an associative array of data set from a SkinTemplate-based
@@ -66,6 +88,12 @@
 		wfSuppressWarnings();
 
 		$this->html( 'headelement' );
+
+		echo "\n<!-- FUSIONFORGE BodyHeader BEGIN -->\n";
+		$GLOBALS['HTML']->bodyHeader($this->params);
+		echo "<div id=\"ff-mw-wrapper\"><div style=\"font-size:x-small;\">\n";
+		echo "<!-- FUSIONFORGE BodyHeader END -->\n";
+
 ?><div id="globalWrapper">
 <div id="column-content"><div id="content">
 	<a id="top"></a>
@@ -162,8 +190,8 @@
 </div>
 <?php
 		$this->printTrail();
-		echo Html::closeElement( 'body' );
-		echo Html::closeElement( 'html' );
+		echo "</div></div>\n";
+		$GLOBALS['HTML']->footer($this->params);
 		wfRestoreWarnings();
 	} // end of execute() method
 
@@ -174,6 +202,30 @@
 		if ( !isset( $sidebar['TOOLBOX'] ) ) $sidebar['TOOLBOX'] = true;
 		if ( !isset( $sidebar['LANGUAGES'] ) ) $sidebar['LANGUAGES'] = true;
 
+		if (session_loggedin()) {
+			$efbox_log_text = _('Log Out');
+			$efbox_log_link = "/account/logout.php?return_to=" .
+			    urlencode(getStringFromServer('REQUEST_URI'));
+		} else {
+			$efbox_log_text = _('Log In');
+			$efbox_log_link = "/account/login.php?return_to=" .
+			    urlencode(getStringFromServer('REQUEST_URI'));
+		}
+		$efbox_prj_link = "/projects/" .
+		    $this->project->getUnixName() . "/";
+		$this->customBox("FusionForge", array(
+			"project" => array(
+				"id" => "fusionforge_project",
+				"href" => $efbox_prj_link,
+				"text" => _("Project Summary"),
+			    ),
+			"loginout" => array(
+				"id" => "fusionforge_loginout",
+				"href" => $efbox_log_link,
+				"text" => $efbox_log_text,
+			    ),
+		    ));
+
 		foreach( $sidebar as $boxName => $content ) {
 			if ( $content === false )
 				continue;
